### PR TITLE
sidebar

### PR DESCRIPTION
**Pull request by pawelbartosik**
first version of sidebar to check

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -27,6 +30,51 @@ function App() {
         </p>
       </div>
       <p className="read-the-docs">Click on the Vite and React logos to learn more</p>
+      <div>
```
Comment (Safeheaded) [inline client/src/App.tsx:33]: Delete the initial, pregenerated content. We don't need it and honestly, it only ruins the experience

File: client/src/components/Layout/Sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const Sidebar: React.FC = () => {
+  const userName: string = 'superadmin';
+
+  return (
+    <Layout className={`${styles.sidebar} ${styles['sidebar-layout']}`}>
+      <Header className={`${styles.sidebar} ${styles['sidebar-header']}`}>
+        <img src="/src/assets/wospLogo.png" alt="WOSP Logo" />
+      </Header>
+      <Layout className={`${styles.sidebar} ${styles['sidebar-content']}`}>
+        <a href="">Strona Główna</a>
```
Comment (Safeheaded) [inline client/src/components/Layout/Sidebar.tsx:16]: Change <a> tags to react-router <NavLink /> tags. Also add detection of currently active page (styles like on hover). Here are the docs: https://reactrouter.com/en/main/components/nav-link

File: client/src/components/Layout/Sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import React from 'react';
+import styles from './Sidebar.module.css';
+
+const { Header, Footer } = Layout;
+
+const Sidebar: React.FC = () => {
+  const userName: string = 'superadmin';
+
+  return (
+    <Layout className={`${styles.sidebar} ${styles['sidebar-layout']}`}>
```
Comment (Safeheaded) [inline client/src/components/Layout/Sidebar.tsx:11]: To apply multiple classes it's better to use this syntax: className={[styles.classOne, styles.classTwo].join(' ')}. And better use camelCase in classes names - you avoid the styles['class-name'] syntax.

File: client/src/components/Layout/Sidebar.tsx (hunk 2 above)
Comment (Safeheaded) [inline client/src/components/Layout/Sidebar.tsx:16]: Change tags to react-router tags. Also add detection of currently active page (styles like on hover). Here are the docs: https://reactrouter.com/en/main/components/nav-link

File: client/src/App.tsx (hunk 1 above)
Comment (Safeheaded) [inline client/src/App.tsx:33]: Delete the initial, pregenerated content. We don't need it and honestly, it only ruins the experience

**Review by Safeheaded — CHANGES_REQUESTED** [review]
Overall - good job. Please make specified adjustments. And delete the counter state too. If you run into any problem - ask here or on discord.

**Comment by Now4czyk** [comment]
According to [mockups](https://www.figma.com/file/dHaZSnxQiLIyLVXauwrApR/WO%C5%9AP?node-id=1%3A3) I assume that there should be 2 views of sidebar: for users that are logged in and for these that are not(view with LoginForm). Please, consider adding a property that controlls switching between these two.

File: client/src/components/Layout/Sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { Button, Layout } from 'antd';
+import React, { Fragment } from 'react';
+
+import styles from './Sidebar.module.css';
+import SidebarItem from './SidebarItem';
+
+const { Header, Footer } = Layout;
+
+const bookmarks = [
```
Comment (Safeheaded) [inline client/src/components/Layout/Sidebar.tsx:9]: Misleading names - these are links, not bookmarks

File: client/src/components/Layout/Sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  { name: 'Strona Główna', url: '#home' },
```
Comment (Safeheaded) [inline client/src/components/Layout/Sidebar.tsx:10]: Delete '#' from urls. And it's a good idea to click through your changes to be 100% they're working before commiting ;)

File: client/src/components/Layout/Sidebar.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  { name: 'Administracja', url: '#admin' },
+  { name: 'Puszki', url: '#box' },
+];
+
+const Sidebar: React.FC = (props) => {
+  const userName = 'superadmin';
+  const isLogin = true;
+
+  return (
+    <Fragment>
```
Comment (Safeheaded) [inline client/src/components/Layout/Sidebar.tsx:21]: Would be good idea to reformat this code into two components, due to low readibility. E.g. <ProtectedContent /> and <UnprotectedContent />

File: client/src/components/Layout/SidebarItem.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+import styles from './SidebarItem.module.css';
+
+type Item = {
+  name: string;
+  url: string;
+};
+
+const SidebarItem: React.FC<Item> = (props) => {
+  const location = useLocation();
```
Comment (Safeheaded) [inline client/src/components/Layout/SidebarItem.tsx:12]: No need to use `useLocation`. NavLink has built in way of detecting if link is active. Read the [docs](https://reactrouter.com/en/main/components/nav-link) and if any questions appear - hit me up

**Review by Safeheaded — CHANGES_REQUESTED** [review]
Almost there! Minor fixes left and we're good to go. And please, delete package-lock.json and yarn.lock from the main directory

File: client/src/components/Layout/Sidebar.tsx (hunk 6 above)
Comment (Safeheaded) [reply client/src/components/Layout/Sidebar.tsx:21]: Github has eaten part of my comment. Here's how it should have been: E.g. `<ProtectedContent />` and `<UnprotectedContent />`.

**Review by Safeheaded — APPROVED** [review]
Well done! Can finally merge!

**Pull request merged**